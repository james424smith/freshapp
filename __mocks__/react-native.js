import * as ReactNative from "react-native";

export * from "react-native";
export const Vibration = { vibrate: jest.fn() };
export const Adnimated = { spring: jest.fn() };

export default Object.setPrototypeOf(
  {
    Vibration,
    Adnimated,
  },
  ReactNative
);
