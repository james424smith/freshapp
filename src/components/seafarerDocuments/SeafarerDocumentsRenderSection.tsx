import * as React from "react";
import { View, TouchableOpacity } from "react-native";
import Text from "../StyledText";
import { Document, PayslipRecord } from "../../interfaces";
import styles from "../../styles/more_screens/SeafarerDocumentationStyles";
import Toggle from "../Toggle";
import {
  getSeafarerDocumentsFile,
  removeDocumentFromLocal,
} from "../../redux/actions";
import { IRootReducerType } from "../../redux/reducers";
import AboutToExpire from "../../../assets/icons/more_screen/seafarerDocumentsScreen/about-to-expire.svg";
import Attached from "../../../assets/icons/more_screen/seafarerDocumentsScreen/attach-icon.svg";
import { useSelector, useDispatch } from "react-redux";
import { useTheme } from "@react-navigation/native";
import * as ShareFileHelper from "../../common/offline_utilities/ShareFiles";
import { widthPercentageToDP as wp } from "react-native-responsive-screen";
import ShareButton from "../../../assets/icons/more_screen/share.svg";
import Colors from "../../constants/styles/colors";
type Props = {
  navigate: (arg1: string, arg2: unknown) => void;
  documentDetails: Document;
};

const handleOnPress = (
  documentDetailsLocal: Document,
  navigate: (arg: string, arg2: { documentDetails: Document }) => void
) => {
  if (
    documentDetailsLocal.documentExtension?.toLowerCase() === "xls" ||
    documentDetailsLocal.documentExtension?.toLowerCase() === "xlsx"
  ) {
    navigate("SeafarerDocumentsExcelFile", {
      documentDetails: documentDetailsLocal,
    });
  } else {
    navigate("SeafarerDocumentsFile", {
      documentDetails: documentDetailsLocal,
    });
  }
};

const Warning = ({
  isExpired,
  isMarked,
}: {
  isExpired: boolean;
  isMarked: boolean;
}) => {
  if (isExpired || isMarked) {
    return <AboutToExpire style={styles.svgSize} fill="#FF0000" />;
  } else {
    return null;
  }
};

const AttachmentIcon = ({ documentDetails }: { documentDetails: Document }) => {
  const theme = useTheme();
  if (documentDetails.hasAttachment && documentDetails.isDeactivated) {
    return (
      <Attached style={styles.svgSize} fill={theme.colors.blackAndWhite} />
    );
  }
  return <Attached style={styles.svgSize} fill={Colors.black} />;
};

const SeafarerDocumentsRenderSection = (props: Props) => {
  const isDownloading = useSelector<IRootReducerType, boolean>(
    ({ seafarerDocuments }) => seafarerDocuments.isDownloading
  );
  const downloadingDocId = useSelector<IRootReducerType, number>(
    ({ seafarerDocuments }) => seafarerDocuments.downloadingDocId
  );
  const downloadingDocCounter = useSelector<IRootReducerType, number>(
    ({ seafarerDocuments }) => seafarerDocuments.downloadingDocCounter
  );
  const dispatch = useDispatch();
  const { navigate, documentDetails } = props;
  const shareIconSize = wp("4%");
  const handleClick = async (
    documentDetailsLocal: Document | PayslipRecord,
    shouldDownload: boolean
  ) => {
    shouldDownload
      ? dispatch(
          getSeafarerDocumentsFile(documentDetailsLocal as Document, true)
        )
      : dispatch(removeDocumentFromLocal(documentDetailsLocal as Document));
  };

  const shouldRenderToggle = (documentDetailsLocal: Document) => {
    return (
      documentDetailsLocal.hasAttachment &&
      documentDetailsLocal.documentExtension?.toLowerCase() !== "xls" &&
      documentDetailsLocal.documentExtension?.toLowerCase() !== "xlsx" && (
        <Toggle
          type="Document"
          documentId={documentDetailsLocal.documentId}
          documentCounter={documentDetailsLocal.documentCounter}
          popUpTitle={"Download Document"}
          documentsFile={documentDetails}
          handleOnPress={handleClick}
          isDownloading={isDownloading}
          downloadingDocId={downloadingDocId}
          downloadingDocCounter={downloadingDocCounter}
        />
      )
    );
  };

  const theme = useTheme();

  const isExpiredOrDeactivated =
    (documentDetails.isExpired && documentDetails.marked) ||
    documentDetails.isDeactivated;

  return (
    <View style={styles.root}>
      <View style={styles.outerView}>
        <View
          style={
            documentDetails.isDeactivated
              ? {
                  ...styles.isDeactivatedElevation,
                  backgroundColor: theme.colors.primary,
                }
              : { ...styles.elevation, backgroundColor: theme.colors.primary }
          }
        >
          <View
            style={
              documentDetails.isOffline
                ? styles.isOfflineContainerColor
                : {
                    ...styles.isOnlineContainerColor,
                    backgroundColor: theme.colors.primary,
                  }
            }
          >
            <View
              style={
                documentDetails.isDeactivated
                  ? styles.isDeactivatedContainerColor
                  : {
                      ...styles.isOnlineContainerColor,
                      backgroundColor: theme.colors.primary,
                    }
              }
            >
              <View style={styles.documentContainer}>
                <TouchableOpacity
                  testID={"document-navigate-button"}
                  disabled={!documentDetails.hasAttachment}
                  onPress={() => {
                    handleOnPress(documentDetails, navigate);
                  }}
                >
                  <View style={styles.makeRow}>
                    <View style={styles.markedContainer}>
                      <AttachmentIcon documentDetails={documentDetails} />
                    </View>
                    <View style={styles.firstRowContainer}>
                      <Text
                        style={
                          documentDetails.isDeactivated
                            ? styles.isDeactivatedTextDescription
                            : {
                                ...styles.textDescription,
                                color:
                                  theme.colors.inactiveTouchableOpacityStyle,
                              }
                        }
                      >
                        {documentDetails.description}
                      </Text>
                    </View>
                    <View style={styles.shouldRenderToggleContainer}>
                      {shouldRenderToggle(documentDetails)}
                    </View>
                  </View>
                  <View style={styles.makeRow}>
                    <View style={styles.markedContainer} />
                    <View style={styles.documentDetailsContainer}>
                      <Text
                        style={
                          documentDetails.isDeactivated
                            ? styles.textDocLicNumber
                            : {
                                ...styles.textDocLicNumber,
                                color:
                                  theme.colors.inactiveTouchableOpacityStyle,
                              }
                        }
                      >
                        {documentDetails.docLicNumber}
                      </Text>
                    </View>
                  </View>
                  <View style={styles.makeRow}>
                    <View style={styles.markedContainer} />
                    <View style={styles.documentDetailsContainer}>
                      <Text
                        style={
                          documentDetails.isDeactivated
                            ? styles.isDeactivatedTextNation
                            : {
                                ...styles.textNation,
                                color:
                                  theme.colors.inactiveTouchableOpacityStyle,
                              }
                        }
                      >
                        {documentDetails.nation}
                      </Text>
                    </View>
                  </View>

                  <View style={styles.makeRow}>
                    <View style={styles.markedContainer}>
                      <Warning
                        isExpired={documentDetails.isExpired}
                        isMarked={documentDetails.isExpired}
                      />
                    </View>

                    <View style={styles.textContainer}>
                      <Text
                        style={[
                          isExpiredOrDeactivated
                            ? styles.dateText
                            : {
                                ...styles.dateTextActive,
                                color: theme.colors.darkNavyToWhite,
                              },
                        ]}
                      >
                        Exp.: {documentDetails.expiryDate}
                      </Text>
                    </View>
                    <View style={styles.issuedDateContainer}>
                      <Text
                        style={
                          isExpiredOrDeactivated
                            ? styles.dateTextIssued
                            : {
                                ...styles.dateTextActiveIssued,
                                color: theme.colors.blackAndWhite,
                              }
                        }
                      >
                        Issued: {documentDetails.issueDate}
                      </Text>
                    </View>
                    <View style={styles.shareButtonContainer}>
                      <View style={styles.shareButtonStyle}>
                        <TouchableOpacity
                          style={styles.shareButtonStyle}
                          testID={"share-document-button"}
                          onPress={async () => {
                            await ShareFileHelper.sharePDF(
                              documentDetails.documentId,
                              documentDetails.documentCounter
                            );
                          }}
                        >
                          <ShareButton
                            fill={
                              isExpiredOrDeactivated
                                ? Colors.black
                                : theme.colors.blackAndWhite
                            }
                            width={shareIconSize}
                          />
                        </TouchableOpacity>
                      </View>
                    </View>
                  </View>
                </TouchableOpacity>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  );
};

export default SeafarerDocumentsRenderSection;
