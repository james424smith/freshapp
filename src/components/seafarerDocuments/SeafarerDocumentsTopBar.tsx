import React from "react";
import { View, TouchableOpacity, ScrollView } from "react-native";
import Text from "../StyledText";
import { Categories } from "../../interfaces";
import styles from "../../styles/more_screens/SeafarerDocumentationStyles";
import { useTheme, Theme } from "@react-navigation/native";
import valueOrDefault from "../../common/valueOrDefault";

type Props = {
  categories?: Categories[];
  handleDocumentsToShow: (category: string) => void;
  selectedCategory: string;
};

function setStyleOfButton(category: string, selectedCategory: string) {
  if (category === selectedCategory) {
    return styles.activeOpacity;
  } else {
    return styles.inactiveOpacity;
  }
}
function setStyleOfText(
  theme: Theme,
  category: string,
  selectedCategory: string
) {
  return category === selectedCategory
    ? {
        ...styles.activeTouchableOpacityLabel,
        color: theme.colors.newAquaMarineColor,
      }
    : {
        ...styles.inactiveTouchableOpacityLabel,
        color: theme.colors.inactiveTouchableOpacityStyle,
      };
}

const SeafarerDocumentsTopBar = (props: Props) => {
  const { categories, handleDocumentsToShow, selectedCategory } = props;
  const theme = useTheme();
  return (
    <ScrollView horizontal={true}>
      <View
        style={{ ...styles.root, backgroundColor: theme.colors.background }}
      >
        <View style={styles.topContainer}>
          {(valueOrDefault(categories, []) as Categories[]).map(
            (category, i) => (
              <TouchableOpacity
                testID={`category-button-${i}`}
                style={[
                  styles.buttonStyle,
                  setStyleOfButton(category.id, selectedCategory),
                ]}
                key={i}
                disabled={category.id === selectedCategory}
                onPress={() => {
                  handleDocumentsToShow(category.id);
                }}
              >
                <Text
                  style={[setStyleOfText(theme, category.id, selectedCategory)]}
                >
                  {category.label}
                </Text>
              </TouchableOpacity>
            )
          )}
        </View>
        <View style={styles.topSeparateLine} />
      </View>
    </ScrollView>
  );
};

export default SeafarerDocumentsTopBar;
