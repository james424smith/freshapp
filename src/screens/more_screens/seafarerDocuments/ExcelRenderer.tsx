import React, { useState, useEffect } from "react";
import { View, Platform } from "react-native";
import { Document, SeafarerDocumentsFile } from "../../../interfaces";
import OpenFile from "@quinaryio/react-native-doc-viewer";
import LoadingScreen from "../../../components/LoadingScreen";
import styles from "../../../styles/more_screens/SalaryPdfStyles";
import { getSeafarerDocumentsFile } from "../../../redux/actions";
import { useSelector, useDispatch } from "react-redux";
import { IRootReducerType } from "../../../redux/reducers";

import { IOS } from "../../../constants/Platforms";
import Text from "../../../components/StyledText";
import { RouteProp } from "@react-navigation/native";
import { DocumentsStackParamList } from "../../../navigation/MoreScreenNavigation";
const logoSize = 100;

interface Props {
  route: RouteProp<DocumentsStackParamList, "SeafarerDocumentsExcelFile">;
}

const ExcelRendererComponent = (props: Props) => {
  const { route } = props;
  const seafarerDocumentsFile = useSelector<
    IRootReducerType,
    SeafarerDocumentsFile | undefined
  >(({ seafarerDocuments }) => seafarerDocuments.seafarerDocumentsFile);
  const isLoadingNew = useSelector<IRootReducerType, boolean>(
    ({ seafarerDocuments }) => seafarerDocuments.isLoadingNew
  );
  const dispatch = useDispatch();
  const [docDetails, setDocumentDetails] = useState<Document | undefined>(
    undefined
  );
  const [animating, setAnimating] = useState(false);
  const [error, setError] = useState<string>("");

  useEffect(() => {
    const documentDetails = route.params?.documentDetails;
    setDocumentDetails(documentDetails);
    dispatch(getSeafarerDocumentsFile(documentDetails, false));
  }, [route.params, dispatch]);

  useEffect(() => {
    const handleError = (errorLocal: any) => {
      if (errorLocal) {
        setAnimating(false);
        setError("error");
      } else {
        setAnimating(false);
      }
    };

    setAnimating(true);
    if (seafarerDocumentsFile && docDetails && !isLoadingNew) {
      if (Platform.OS === IOS) {
        OpenFile.openDocb64(
          [
            {
              base64: seafarerDocumentsFile.document,
              fileName: `doc-${docDetails.documentId}-counter-${
                docDetails.documentCounter
              }.${seafarerDocumentsFile.fileType.toLowerCase()}`,
              fileType: seafarerDocumentsFile.fileType.toLowerCase(),
            },
          ],
          (errorLocal: any, _url: any) => {
            handleError(errorLocal);
          }
        );
      } else {
        OpenFile.openDocb64(
          [
            {
              base64: seafarerDocumentsFile.document,
              fileName: `doc-${docDetails.documentId}-counter-${
                docDetails.documentCounter
              }.${seafarerDocumentsFile.fileType.toLowerCase()}`,
              fileType: seafarerDocumentsFile.fileType.toLowerCase(),
              // eslint-disable-next-line @typescript-eslint/ban-ts-comment
              // @ts-ignore
              cache: false,
            },
          ],
          (errorLocal: any, _url: any) => {
            handleError(errorLocal);
          }
        );
      }
    }
  }, [seafarerDocumentsFile, docDetails, isLoadingNew]);

  if (error) {
    return (
      <View style={styles.root}>
        <Text>
          Your phone cannot open {seafarerDocumentsFile?.fileType} files. Please
          download an app that is capable of opening such files and try again.
        </Text>
      </View>
    );
  } else if (animating || isLoadingNew) {
    return (
      <View style={styles.root}>
        <LoadingScreen logoWidthSize={logoSize} logoHeightSize={logoSize} />
      </View>
    );
  } else {
    return (
      <View style={styles.root}>
        <Text>
          {docDetails?.description} is open in an external{" "}
          {docDetails?.documentExtension?.toLowerCase()} app.
        </Text>
      </View>
    );
  }
};

export default ExcelRendererComponent;
