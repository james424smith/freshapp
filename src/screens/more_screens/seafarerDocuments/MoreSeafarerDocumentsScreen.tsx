import React, { useState, useEffect } from "react";
import { ScrollView, StyleSheet, View } from "react-native";
import { SeafarerDocumentsFile } from "../../../interfaces";
import { useSelector, useDispatch } from "react-redux";
import { getSeafarerDocumentsFile } from "../../../redux/actions";
import ImageRender from "../../../components/ImgRender";
import * as RNFS from "react-native-fs";
import LoadingScreen from "../../../components/LoadingScreen";

import { IRootReducerType } from "../../../redux/reducers";
import PdfRender from "../../../components/PdfRender";
import { DocumentsStackParamList } from "../../../navigation/MoreScreenNavigation";
import { RouteProp } from "@react-navigation/native";

interface Props {
  route: RouteProp<DocumentsStackParamList, "SeafarerDocumentsFile">;
}

const styles = StyleSheet.create({
  scrollViewContainer: {
    flexGrow: 1,
  },
  scrollView: {
    flex: 1,
  },
  root: { flex: 1 },
  container: { alignSelf: "center", marginTop: 50 },
  centerIndicator: {
    flex: 1,
    marginBottom: 50,
    alignItems: "center",
    justifyContent: "center",
  },
});
const logoSize = 100;

const SeafarerDocumentsFileScreen = (props: Props) => {
  const [documentFromLocal, setDocumentFromLocal] = useState<
    SeafarerDocumentsFile | undefined
  >(undefined);
  const { route } = props;

  const seafarerDocumentsFile = useSelector<
    IRootReducerType,
    SeafarerDocumentsFile | undefined
  >(({ seafarerDocuments }) => seafarerDocuments.seafarerDocumentsFile);
  const isLoadingNew = useSelector<IRootReducerType, boolean>(
    ({ seafarerDocuments }) => seafarerDocuments.isLoadingNew
  );
  const dispatch = useDispatch();
  const openFileFromLocalStorage = (
    documentId: string | number,
    documentCounter: number,
    fileType: string
  ) => {
    return `${RNFS.DocumentDirectoryPath}/document/id-${documentId}-counter-${documentCounter}.${fileType}`;
  };

  useEffect(() => {
    const documentDetails = route.params?.documentDetails;
    //if document switched to offline
    if (documentDetails.isOffline) {
      const { documentId, documentCounter, fileType } = documentDetails;
      //open the offline document from the path
      RNFS.readFile(
        openFileFromLocalStorage(documentId, documentCounter, fileType)
      )
        .then((file: any) => {
          setDocumentFromLocal({ document: file, fileType });
        })
        .catch((error: any) => {
          console.log(error);
        });
      //If not offline, fetch the document from API
    } else {
      dispatch(getSeafarerDocumentsFile(documentDetails, false));
    }
  }, [dispatch, route.params]);

  const chooseComponentToShow = (file: SeafarerDocumentsFile) => {
    return file.fileType === "PDF" ? (
      <PdfRender document={file.document} resourceType={"base64"} />
    ) : (
      <ImageRender
        document={file.document}
        resourceType={"base64"}
        fileType={file.fileType}
      />
    );
  };

  const itemToShow = (
    isLoadingNewLocal: boolean,
    imgFile?: SeafarerDocumentsFile
  ) => {
    if (isLoadingNewLocal) {
      return (
        <View style={styles.root}>
          <LoadingScreen logoWidthSize={logoSize} logoHeightSize={logoSize} />
        </View>
      );
    } else {
      return imgFile ? chooseComponentToShow(imgFile) : undefined;
    }
  };

  //checking if the document is offline.
  const img = documentFromLocal ? documentFromLocal : seafarerDocumentsFile;
  return (
    <ScrollView
      style={styles.scrollView}
      contentContainerStyle={styles.scrollViewContainer}
    >
      {itemToShow(isLoadingNew, img)}
    </ScrollView>
  );
};

export default SeafarerDocumentsFileScreen;
