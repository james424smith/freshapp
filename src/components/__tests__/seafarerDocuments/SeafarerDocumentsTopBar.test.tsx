import React from "react";
import SeafarerDocumentsTopBar from "../../seafarerDocuments/SeafarerDocumentsTopBar";
import { categories } from "../../../../__mocks__/fake-data/seafarerDocuments.json";
import { render, cleanup, fireEvent } from "@testing-library/react-native";

describe("SeafarerDocumentsTopBar snapshot test", () => {
  afterEach(() => {
    cleanup();
    jest.clearAllMocks();
  });

  it("should render correct ui when allotmentTypeText", () => {
    const handleDocumentsToShow = jest.fn();
    const { toJSON, getByTestId } = render(
      <SeafarerDocumentsTopBar
        handleDocumentsToShow={handleDocumentsToShow}
        selectedCategory={"1"}
        categories={categories}
      />
    );

    fireEvent.press(getByTestId("category-button-0"));
    expect(handleDocumentsToShow).toHaveBeenCalledWith("3");
    expect(toJSON()).toMatchSnapshot();
  });
});
