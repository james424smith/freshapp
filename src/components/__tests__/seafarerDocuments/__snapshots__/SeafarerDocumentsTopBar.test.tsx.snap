// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SeafarerDocumentsTopBar snapshot test should render correct ui when allotmentTypeText 1`] = `
<RCTScrollView
  horizontal={true}
>
  <View>
    <View
      style={
        Object {
          "backgroundColor": "rgb(242, 242, 242)",
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignContent": "center",
            "alignItems": "flex-start",
            "flexDirection": "row",
            "flexWrap": "nowrap",
            "justifyContent": "space-around",
            "marginTop": 22.5,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "opacity": 0.5,
              "padding": 30,
            }
          }
          testID="category-button-0"
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#2A2E43",
                  "fontFamily": "Roboto-Regular",
                },
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 22.5,
                    "fontWeight": "bold",
                    "opacity": 0.5,
                  },
                ],
              ]
            }
          >
            Travel/ID/Medical/Other
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "opacity": 1,
              "padding": 30,
            }
          }
          testID="category-button-1"
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#2A2E43",
                  "fontFamily": "Roboto-Regular",
                },
                Array [
                  Object {
                    "borderBottomColor": "#0093BB",
                    "borderBottomWidth": 2,
                    "color": undefined,
                    "fontSize": 26.5,
                  },
                ],
              ]
            }
          >
            STCW National
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "opacity": 0.5,
              "padding": 30,
            }
          }
          testID="category-button-2"
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#2A2E43",
                  "fontFamily": "Roboto-Regular",
                },
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 22.5,
                    "fontWeight": "bold",
                    "opacity": 0.5,
                  },
                ],
              ]
            }
          >
            Flag State
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "opacity": 0.5,
              "padding": 30,
            }
          }
          testID="category-button-3"
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#2A2E43",
                  "fontFamily": "Roboto-Regular",
                },
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 22.5,
                    "fontWeight": "bold",
                    "opacity": 0.5,
                  },
                ],
              ]
            }
          >
            Non STCW
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "opacity": 0.5,
              "padding": 30,
            }
          }
          testID="category-button-4"
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#2A2E43",
                  "fontFamily": "Roboto-Regular",
                },
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 22.5,
                    "fontWeight": "bold",
                    "opacity": 0.5,
                  },
                ],
              ]
            }
          >
            Company Specific Training
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "borderBottomColor": "#78849E",
            "borderBottomWidth": 0.5,
            "marginTop": 15,
          }
        }
      />
    </View>
  </View>
</RCTScrollView>
`;
